gcc/rust/lex/rust-lex.cc

```diff
@@ -186,6 +202,50 @@ Lexer::skip_input ()
   skip_input (0);
 }
 
+void
+Lexer::skip_token (int n)
+{
+  // dump tokens if dump-lex option is enabled
+  if (dump_lex_enabled && !dump_lex_out.fail ())
+    {
+      bool found_eof = false;
+      const_TokenPtr tok;
+      for (int i = 0; i < n + 1; i++)
+	{
+	  if (!found_eof)
+	    {
+	      tok = peek_token ();
+	      found_eof |= tok->get_id () == Rust::END_OF_FILE;
+	      bool has_text = tok->get_id () == Rust::IDENTIFIER
+			      || tok->get_id () == Rust::INT_LITERAL
+			      || tok->get_id () == Rust::FLOAT_LITERAL
+			      || tok->get_id () == Rust::STRING_LITERAL
+			      || tok->get_id () == Rust::CHAR_LITERAL
+			      || tok->get_id () == Rust::BYTE_STRING_LITERAL
+			      || tok->get_id () == Rust::BYTE_CHAR_LITERAL;
+
+	      Location loc = tok->get_locus ();
+
+	      dump_lex_out << "<id=";
+	      dump_lex_out << tok->token_id_to_str ();
+	      dump_lex_out << (has_text
+				 ? (std::string (", text=") + tok->get_str ()
+				    + std::string (", typehint=")
+				    + std::string (tok->get_type_hint_str ()))
+				 : "")
+			   << " ";
+	      dump_lex_out << get_line_map ()->to_string (loc) << " ";
+	    }
+
```

Comment:
This should be in a separate function, so that we could have something like this:

```cpp
if (dump_lex_enabled)
    dump_until_n();
   
token_queue.skip(n);
```

Reply:
fixed